### PR TITLE
adds appropriate checks for new social links

### PR DESCRIPTION
**Pull request by acthomas06**
This will add tests for social links for: steam, instagram, soundcloud, spreadshirt, and patreon.

### CODE REVIEW

File: lib/is.js [hunk 1]
```diff
@@ -71,6 +71,81 @@ var util = require('./util');
...
+ */
+(function () {
+    var identifier = '[a-z0-9\\-_]{1,100}';
+    def({
+        fn: 'patreon.name',
+        re: new RegExp('^' + identifier + '$', 'i')
+    });
+    def({
+        fn: 'patreon.url',
+        re: url('patreon\\.com\\/.+' + identifier)
```
Comment (connor4312) [inline lib/is.js]: Ending these with `.+[a-z0-9\\-_]{1,100}` is probably not what you want. It'll match anything that ends in a-z0-9-_, and is backtracking

**Comment by connor4312** [comment]
Tests please, along with that fix :)

File: lib/is.js (hunk 1 above)
Comment (acthomas06) [inline lib/is.js]: Which pattern do you suggest?  My regex is iffy!

Comment (connor4312) [inline lib/is.js]: Getting rid of the .+ in this and all above cases should work

**Pull request merged**